lib/suspenders/generators/stylelint_generator.rb

```diff
@@ -0,0 +1,74 @@
+require_relative "base"
+
+module Suspenders
+  class StylelintGenerator < Generators::Base
+    def setup_hound
+      action InvokeGenerator.new(self, "suspenders:lint")
+      action ToggleComments.new(self, ".hound.yml", /stylelintrc/)
+    end
+
+    def install_stylelint
+      dependencies = ["stylelint", "@thoughtbot/stylelint-config"]
+      action YarnInstall.new(self, dependencies, "--dev")
+    end
+
+    def copy_stylelint_config
+      copy_file "stylelintrc.json", ".stylelintrc.json"
+    end
+
+    class InvokeGenerator
+      def initialize(base, generator)
+        @base = base
+        @generator = generator
+      end
+
+      def invoke!
+        @base.invoke @generator
+      end
+
+      def revoke!; end
+    end
+
+    class ToggleComments
+      def initialize(base, filename, pattern)
+        @base = base
+        @filename = filename
+        @pattern = pattern
+      end
+
+      def invoke!
+        @base.uncomment_lines(@filename, @pattern)
+      end
+
+      def revoke!
+        @base.behavior = :invoke
+        @base.comment_lines(@filename, @pattern)
+      ensure
+        @base.behavior = :revoke
+      end
+    end
+
+    class YarnInstall
+      def initialize(base, dependencies, flags)
+        @base = base
+        @dependencies = dependencies
+        @flags = flags
+      end
+
+      def invoke!
+        @dependencies.each do |dependency|
+          @base.run "bin/yarn add #{dependency} #{@flags}"
```

Comment:
Have you considered running `yarn add` with multiple package names?

```rb
dependencies_argument = @dependencies.join(" ")

@base.run "bin/yarn add #{dependencies_argument} #{@flags}"
```

Reply:
Yeah, that seems reasonable. Maybe I will also be able to get rid of the `reverse` if I do it that way.